### DISCUSSION TITLE
Query all visuals

### DISCUSSION THREAD

**Discussion by pawarbi** [tabulartools/pbi-explorer#9] [Q&A]
Hi, great work and thanks to the team for open-sourcing it. 

Is it possible to query the entire report based on some filter conditions? e.g. Return pages where IsHidden=True or return visuals where DefaultSortOrder=False and number of visual level filters >0 or in a a report cards always need to be aligned with fixed X axis location on all pages, so return pages where card visual x position <> some_value. That would be very helpful to work with inherited reports. Also, just a suggestion, color coding visuals would be great too. e.g. I may have 20 visuals on a page (I won't but inherited :D) and if the visuals could be highlighted with a red border or something so its easier to spot when certain conditions are met. 

Thanks again, this is going to be a massive help.

**Comment by DBojsen** [comment]
Hi @pawarbi 

We'll soon introduce the Report Analyzer (like Tabular Editors Best Practice Analyzer), which should bring solutions to all of this.
So please be patient and try out the coming beta versions.

BR David - Tabular Tools
